doc/coding_standard_c.md

```diff
@@ -163,11 +163,65 @@ within the source even if they are already included in that header. This
 provides a complete context for readers of the source file... i.e., they
 don't have to search through headers to determine where a resource came from.
 
+Files included by all source files must conform to the following format and
+order. Each entry in the list below defines a contiguous block of `include`
+directives separated by a blank line:
+* System headers - These are headers provided by the core c libraries
+(typically from libc).
+* External dependencies - These are headers installed on the platform defining
+interfaces to external libraries.
+* Standard TSS2 headers - These are headers that define the public TSS2 types
+and interfaces. They are all located under $(srcdir)/include/* and will be
+installed as part of the `install` build target. These *must* be included
+using the quoted include variant (using `"` instead of the angle brackets).
+* Internal headers - These are headers defining the interfaces to code modules
+that are internal to the project.
+
+Headers in each block must listed in alphabetical order.
+
 ### Example
-Consider a header and source that use FILE pointers declared in stdio.h. The
-header must `#include <stdio.h>` to ensure files that include it will be able
-to compile. The corresponding source #includes the associated header and
-therefore receives the benefit of stdio.h automatically.
+header: `example.h`
+```
+/*
+ * BSD license block
+ */
+#ifndef HEADER_GUARD_H
+#define HEADER_GUARD_H
+
+#include <stdint.h>
+#include <stdio.h>
```

Comment:
stdio.h is a bad example here. No one should need it in TSS.

Reply:
I get your point but no one should need `foo/bar.h` from further down in the example either :stuck_out_tongue:

I guess my point is that this is a slippery slope: The header guard doesn't match the file name. The headers for internal modules aren't real headers ... I'm just not sure where the bottom will be though.

Are you willing to limit your requested change to picking a more realistic header instead of `stdio.h` or do you intend for this to be more far-reaching?

Reply:
cool thanks. Just fixed this up. Merging once I get a clean build